src/lib.rs

```diff
@@ -237,7 +245,9 @@ impl Config {
                 // On MinGW we need to coerce cmake to not generate a visual
                 // studio build system but instead use makefiles that MinGW can
                 // use to build.
-                cmd.arg("-G").arg("MSYS Makefiles");
+                if self.generator.is_none() {
+                    self.generator("MSYS Makefiles");
```

Comment:
Could this actually just pass down `-G` directly? In theory we don't want to do *too* much mutation as part of this call of the `Config` structure itself